forge-ai/src/main/java/forge/ai/ComputerUtilCard.java

```diff
@@ -762,29 +762,17 @@ public static String getMostProminentCardName(final CardCollectionView list) {
         return maxName;
     }
 
-    public static String getMostProminentBasicLandType(final CardCollectionView list) {
-        return getMostProminentType(list, CardType.getBasicTypes());
+    public static String getMostProminentType(final CardCollectionView list, final Collection<String> valid) {
+        return getMostProminentType(list, valid, true);
     }
 
-    /**
-     * <p>
-     * getMostProminentCreatureType.
-     * </p>
-     * 
-     * @param list
-     * @return a {@link java.lang.String} object.
-     */
-    public static String getMostProminentCreatureType(final CardCollectionView list) {
-        return getMostProminentType(list, CardType.getAllCreatureTypes());
-    }
-    public static String getMostProminentType(final CardCollectionView list, final Collection<String> valid) {
+    public static String getMostProminentType(final CardCollectionView list, final Collection<String> valid, boolean includeTokens) {
         if (list.size() == 0) {
             return "";
         }
 
         final Map<String, Integer> typesInDeck = Maps.newHashMap();
 
-        // TODO JAVA 8 use getOrDefault
         for (final Card c : list) {
             // Changeling are all creature types, they are not interesting for
```

Comment:
You'll probably also want to check if c is a token here to ignore all those that have already been created in play (`getAllCards` should include those)

Reply:
Good call, done!